server/eventlog/eventlog.go

```diff
@@ -24,40 +37,194 @@ func GetEventLogChunk(ctx context.Context, env environment.Env, req *elpb.GetEve
 	}
 
 	c := chunkstore.New(env.GetBlobstore(), &chunkstore.ChunkstoreOptions{})
+	eventLogPath := GetEventLogPathFromInvocationId(req.InvocationId)
 	var intChunkId uint16
 	if len(req.ChunkId) > 0 {
-		n, err := strconv.ParseUint(req.ChunkId, 16, 16)
-		if err != nil {
+		var err error
+		if intChunkId, err = chunkstore.ChunkIndexAsUint16(req.ChunkId); err != nil {
 			return nil, err
 		}
-		intChunkId = uint16(n)
-	} else if len(inv.LastChunkId) > 0 {
-		n, err := strconv.ParseUint(inv.LastChunkId, 16, 16)
-		if err != nil {
+	} else {
+		startingIndex, _ := chunkstore.ChunkIndexAsUint16(inv.LastChunkId)
+		var err error
+		if intChunkId, err = c.GetLastChunkIndex(ctx, eventLogPath, startingIndex); err != nil {
 			return nil, err
```

Comment:
If err is NotFound or OutOfRange, then instead of returning the err, this could set `rsp.NextChunkId = ""` (but only if `inv.InvocationStatus != IN_PROGRESS`) and that would let us simplify the client quite a bit

Reply:
Done; is this how you meant for it to be done?

Reply:
Figured this out in direct communication.